integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java

```diff
@@ -242,6 +247,8 @@ public void statusUpdate(SchedulerDriver driver, Protos.TaskStatus status) {
         case TASK_ERROR:
           if (status.getTaskId().getValue().equals(String.valueOf(mMasterTaskId))) {
             mMasterCount --;
+          } else if (status.getTaskId().getValue().equals(String.valueOf(mWorkerTaskId))) {
```

Comment:
@apc99 is right ... you cannot use mWorkerTaskId to identify all possible worker

instead you should be comparing task name against `Constants.INTEGRATION_MESOS_TACHYON_MASTER_NAME` and `Constants.INTEGRATION_MESOS_TACHYON_WORKER_NAME`

At that point, you can get rid of `mMasterTaskId` and `mWorkerTaskId`.